org.opent2t.sample.hub.superpopular/com.insteon.hub/js/thingTranslator.js

```diff
@@ -398,18 +412,22 @@ class Translator {
         return sleep(1000).then(() => {
             return this._makeRequest(this._commandPath + '/' + commandId, 'GET')
                 .then( (response) => {
-                    if(response.status === 'pending')
-                    {
-                        if(count > 1){
-                            return this._getCommandResponse(commandId, --count);
-                        } else {
-                            this._throwError('Internal Error - Insteon command timeout.');
-                        }
-                    } else {
-                        return response;
+                    switch(response.status){
+                        case 'pending':
+                            if(count > 1){
+                                return this._getCommandResponse(commandId, --count);
+                            } else {
+                                this._throwError('Internal Error - Insteon command timeout.');
+                            }
+                            break;
+                        case 'succeeded':
+                            return Promise.resolve(response);
+                        case 'failed':
+                        default:
+                            this._throwError( 'Internal Error - Insteon command failed for deviceId ' + response.command.device_id + '.');
```

Comment:
Yes, that is why I only log the failure for _providerSchemaToPlatformSchema before. However, I do agree that we should let user know there is a communication problem. Should we have another resource to indicate if a device is online/offline?

---
In reply to: [103536052](https://github.com/openT2T/translators/pull/187#discussion_r103536052) [](ancestors = 103536052)